From e6c148fb958ad71e2ebf00b46aafc7c66202372f Mon Sep 17 00:00:00 2001
From: Mike Sarahan <msarahan@gmail.com>
Date: Fri, 27 Sep 2019 10:55:32 -0500
Subject: [PATCH] tweak import error message to link to Numpy MKL
 troubleshooting

---
 Orange/__init__.py | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/Orange/__init__.py b/Orange/__init__.py
index 609da3d..2d033cb 100644
--- a/Orange/__init__.py
+++ b/Orange/__init__.py
@@ -3,10 +3,15 @@
 
 try:
     from Orange.data import _variable
-except ImportError:
-    raise ImportError("Compiled libraries cannot be found.\n"
-                      "Try reinstalling the package with:\n"
-                      "pip install --no-binary Orange3") from None
+except ImportError as e:
+    if sys.platfom == 'win32':
+        raise ImportError("Compiled libraries cannot be loaded.\n"
+                          "For potential causes and workarounds, please see\n"
+                          "https://conda.io/projects/conda/en/latest/user-guide/troubleshooting.html#numpy-mkl-library-load-failed") from None
+    else:
+        raise ImportError("Compiled libraries cannot be found.\n"
+                        "Try reinstalling the package with:\n"
+                        "pip install --no-binary Orange3") from None
 
 from Orange import data
 
-- 
2.20.1

