From 2b1558a1705664c3c456bbaf0c3408a6744c461b Mon Sep 17 00:00:00 2001
From: Ales Erjavec <ales.erjavec@fri.uni-lj.si>
Date: Fri, 22 Dec 2023 10:39:23 +0100
Subject: [PATCH] setup.py: Replace use of imp module

---
 setup.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 230c3f2de..a00d449d5 100755
--- a/setup.py
+++ b/setup.py
@@ -163,8 +163,12 @@ if not release:
         GIT_REVISION = git_version()
     elif os.path.exists('Orange/version.py'):
         # must be a source distribution, use existing version file
-        import imp
-        version = imp.load_source("Orange.version", "Orange/version.py")
+        import importlib.util
+        spec = importlib.util.spec_from_file_location(
+            "Orange.version", filename
+        )
+        version = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(version)
         GIT_REVISION = version.git_revision
     else:
         GIT_REVISION = "Unknown"
--
2.32.1 (Apple Git-133)
